imageregistry/v1/types.go

```diff
@@ -306,6 +306,78 @@ type ImageRegistryConfigStorageIBMCOS struct {
 	ServiceInstanceCRN string `json:"serviceInstanceCRN,omitempty"`
 }
 
+// EndpointAccessibility defines the Alibaba VPC endpoint for storage
+type EndpointAccessibility string
+
+// AlibabaEncryptionMode defines an enumerable type for the encryption mode
+type AlibabaEncryptionMode string
+
+const (
+	// InternalEndpoint sets the VPC endpoint to internal
+	InternalEndpoint EndpointAccessibility = "Internal"
+	// PublicEndpoint sets the VPC endpoint to public
+	PublicEndpoint EndpointAccessibility = "Public"
+
+	// PlainText is an AlibabaEncryptionMode. This means no encryption
+	PlainText AlibabaEncryptionMode = "PlainText"
+	// AES256 is an AlibabaEncryptionMode. This means AES256 encryption
+	AES256 AlibabaEncryptionMode = "AES256"
+	// KMS is an AlibabaEncryptionMode. This means KMS encryption
+	KMS AlibabaEncryptionMode = "KMS"
+)
+
+// EncryptionAlibaba this a union type in kube parlance.  Depending on the value for the encryptionType,
+// different pointers may be used
+type EncryptionAlibaba struct {
+	// EncryptionType defines the different encrytion types available
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default is `PlainText`.
+	// +kubebuilder:validation:Enum="PlainText";"KMS";"AES256"
+	// +kubebuilder:validation:Required
+	Type AlibabaEncryptionMode `json:"type"`
+
+	// KMS (key management service) is an encryption type provided to encrypt the provider
+	// +optional
+	KMS *KMSEncryptionAlibaba `json:"kms,omitempty"`
```

Comment:
if this is only to be set when type matches, you have to add a patch file (can be follow-up), setting
```
anyOf:
- not:
    properties:
      type:
        enum: ["KMS"]
  not:
    required: ["kms"]
- properties:
    type:
      enum: ["KMS"]
   required: ["kms"]
```

Double check the logic.

Reply:
Let me research how to do this.

Reply:
From our discussion in slack https://github.com/openshift/api/pull/1082/files#diff-3e8a362860cb72aadab809e7fdb9a0d112ed3e6f09c50c89ac77cf3453189880